platform/ios/docs/guides/Data-Driven Styling.md

```diff
@@ -0,0 +1,124 @@
+<!---
+To do - JK:
+- [ ] Fill it out with more detailed explanations
+- [ ] Add examples for camera or composite style functions
+- [ ] Cluster earthquakes as an example for camera function?
+- [ ] Clean up & optimize screenshots - replace with screenshots showing circles below symbol layer
+- [ ] Link API docs
+- [ ] Add cocoaprefix so this can be moved to darwin
+---->
+
+##Data-Driven Styling
+
+Mapbox’s data-driven styling features allow you to use data properties to style your maps. You can style objects within the same layer differently based on their individual attributes. This enables you to style icons, routes, parks, and more based on attributes.
+
+![available bikes](img/data-driven-styling/citibikes.png)
+
+_Insert cool overview of things you can do! Try to include polylines & polygons_
+
+### How to use Data-Driven Styling
+This guide uses earthquake data from the [U.S. Geological Survey](https://earthquake.usgs.gov/earthquakes/feed/v1.0/geojson.php) to style a map based on attributes. For more information about how to work with GeoJSON data in our iOS SDK, please see our [working with GeoJSON data](https://www.mapbox.com/ios-sdk/api/3.5.0-beta.1/working-with-geojson-data.html) guide.
+
+`MGLStyleFunction`
+
+There are three subclasses of `MGLStyleFunction`:
+
+* `MGLCameraStyleFunction` - For a style value that changes with zoom level. For example, you can make the radius of a circle to increase based on zoom level. In iOS SDK v3.4, this was a `MGLStyleFunction`.
+* `MGLSourceStyleFunction` - For a style value that changes with the attributes of a feature. You can adjust the radius of a circle based on the magnitude of an earthquake, for example.
+* `MGLCompositeStyleFunction` - For a style value that changes with both zoom level and attribute values. For example, you can add a circle layer where each circle has a radius based on both zoom level and the magnitude of an earthquake.
+
+The documentation for individual style properties will note which style functions are enabled for that property.
+
+####Stops
+
+Stops are key-value pairs that that determine a style value. With a `MGLCameraSourceFunction` stop, you can use a dictionary with a zoom level for a key and a `MGLStyleValue` for the value. For example, you can use a stops dictionary with zoom levels 0, 10, and 20 as keys, and yellow, orange, and red as the values. A `MGLSourceStyleFunction` uses the relevant attribute value as the key.
+
+####`MGLInterpolationMode`
```

Comment:
In the generated documentation, the four `#`s capitalization issue combined with the formatting caused by the back ticks makes this looke like `MGLINTERPOLATIONMODE` with a really thin font weight.